test/simulator/timestamp/src/timestamp_manager.cpp

```diff
@@ -222,3 +222,70 @@ timestamp_manager::validate_read_timestamp(session_simulator *session, const uin
 
     return (0);
 }
+
+/*
+ * Validate the commit timestamp. The constraints on the commit timestamp are:
+ */
+int
+timestamp_manager::validate_commit_timestamp(
+  session_simulator *session, const uint64_t commit_ts) const
+{
+    uint64_t prepare_ts = session->get_prepare_timestamp();
+    if (!prepare_ts) {
+        /*
+         * We cannot set the commit timestamp to be earlier than the fist commit timestamp when
+         * setting the commit timestamp multiple times within a transaction.
+         */
+        uint64_t first_commit_ts = session->get_first_commit_timestamp();
+        if (first_commit_ts != 0 && commit_ts < first_commit_ts) {
+            WT_SIM_RET_MSG(EINVAL,
+              "commit timestamp " + std::to_string((uint64_t)commit_ts) +
```

Comment:
You don't need to cast `commit_ts ` to `uint64_t`. Please check the rest of the code.